Cython/Compiler/ExprNodes.py

```diff
@@ -196,9 +196,9 @@ def make_dedup_key(outer_type, item_nodes):
 
     @param outer_type: The type of the outer container.
     @param item_nodes: A sequence of constant nodes that will be traversed recursively.
-    @return: A tuple that can be used as a dict key for deduplication.
+    @return: A hashable, constant sequence that can be used as a dict key for deduplication.
     """
-    item_keys = [
+    item_keys = tuple(
```

Comment:
List comprehensions are more efficient than generator expressions. This was intentional.
```suggestion
    item_keys = [
```